# The Battle of Neighborhoods (Part I)

#####  By: Aminath Shausan

### Background

### The Problem

### Description of Data and its Application

#### 1.  Data Collection and Preprocesing 

In [6]:
#import required libraries
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

print('Libraries imported.')

Libraries imported.


In [7]:
#create a list of  suburb names of Brisbane
inner_sub = ['Bowen Hills',  'Brisbane', 'East Brisbane', 'Fortitude Valley', 'Herston', 'Highgate Hill', 'Kangaroo Point', 'Kelvin Grove', 
             'New Farm', 'Newstead', 'Paddington', 'Petrie Terrace', 'Red Hill', 'South Brisbane', 'Spring Hill', 'Teneriffe', 'West End', 
             'Woolloongabba']

north_sub = ['Albion', 'Alderley', 'Ascot', 'Aspley', 'Bald Hills', 'Banyo', 'Boondall', 'Bracken Ridge', 'Bridgeman Downs', 'Brighton',
             'Brisbane Airport', 'Carseldine', 'Chermside', 'Chermside West', 'Clayfield',  'Deagon', 'Eagle Farm', 'Everton Park', 'Ferny Grove',
             'Fitzgibbon', 'Gaythorne', 'Geebung', 'Gordon Park', 'Grange', 'Hamilton', 'Hendra', 'Kalinga', 'Kedron', 'Keperra', 'Lutwyche', 
             'McDowall', 'Mitchelton', 'Myrtletown', 'Newmarket', 'Northgate', 'Nudgee', 'Nudgee Beach', 'Nundah', 'Pinkenba', 'Sandgate',
             'Shorncliffe', 'Stafford', 'Stafford Heights', 'Taigum', 'Virginia', 'Wavell Heights', 'Wilston', 'Windsor', 'Wooloowin', 'Zillmere']

south_sub = ['Acacia Ridge', 'Algester', 'Annerley', 'Archerfield', 'Burbank', 'Calamvale', 'Coopers Plains', 'Darra', 'Doolandella', 
             'Drewvale', 'Durack', 'Dutton Park', 'Eight Mile Plains', 'Ellen Grove', 'Fairfield', 'Forest Lake', 'Greenslopes', 'Heathwood', 
             'Holland Park', 'Holland Park West', 'Inala', 'Karawatha', 'Kuraby', 'Larapinta', 'MacGregor', 'Mackenzie', 'Mansfield', 'Moorooka', 
             'Mount Gravatt', 'Mount Gravatt East', 'Nathan', 'Pallara', 'Parkinson', 'Richlands', 'Robertson', 'Rochedale', 'Rocklea', 
             'Runcorn', 'Salisbury', 'Seventeen Mile Rocks',  'Sinnamon Park', 'Stones Corner', 'Stretton', 'Sumner', 'Sunnybank',
             'Sunnybank Hills', 'Tarragindi',  'Tennyson', 'Upper Mount Gravatt', 'Wacol', 'Willawong',  'Wishart', 'Yeerongpilly', 'Yeronga']

east_sub = ['Balmoral', 'Belmont', 'Bulimba', 'Camp Hill', 'Cannon Hill', 'Carina', 'Carina Heights', 'Carindale', 'Chandler', 'Coorparoo', 
            'Gumdale', 'Hawthorne', 'Hemmant', 'Lota', 'Lytton', 'Manly', 'Manly West', 'Moreton Island', 'Morningside', 'Murarrie', 
            'Norman Park', 'Port of Brisbane', 'Ransome', 'Seven Hills', 'Tingalpa', 'Wakerley', 'Wynnum', 'Wynnum West']

west_sub = ['Anstead', 'Ashgrove', 'Auchenflower', 'Bardon', 'Bellbowrie', 'Brookfield', 'Chapel Hill', 'Chelmer', 'Chuwar', 'Corinda', 'Enoggera', 
            'Enoggera Reservoir', 'Fig Tree Pocket', 'Graceville', 'Indooroopilly', 'Jamboree Heights' , 'Jindalee', 'Karana Downs', 
            'Kenmore', 'Kenmore Hills', 'Kholo', 'Lake Manchester', 'Middle Park', 'Milton', 'Moggill', 'Mount Coot-tha', 'Mount Crosby', 
            'Mount Ommaney', 'Oxley', 'Pinjarra Hills', 'Pullenvale', 'Riverhills', 'Seventeen Mile Rocks', 'Sherwood', 'Sinnamon Park', 
            'St Lucia', 'Taringa', 'The Gap', 'Toowong', 'Upper Brookfield', 'Upper Kedron', 'Westlake']

#check length of the list 
#print(len(inner_sub), len(north_sub), len(south_sub), len(east_sub), len(west_sub))

#join the list of suburbs into one list 
sub_list  = inner_sub + north_sub + south_sub + east_sub + west_sub
print(len(sub_list))

192


In [8]:
#urls=['http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=0', 
#      'http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=&page=1']

urls = {page_number:'http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=&page={}'.format(page_number) 
            for page_number in list(range(0, 29))}
#table_dfs= [pd.read_html(url) for url in urls]
dfnew = {}
dfs ={}
for url in range(0,len(urls)):
    dfs[url]= (pd.read_html(urls[url])[5]) 
    dfs[url].columns = ['Rank', 'Suburb', 'Value']
    dfs[url].drop(dfs[url].index[0], inplace=True)
    #dfnew[url] = dfs[url].loc[(dfs[url]['Suburb'].isin(sub_list))] 
    dfnew[url] = dfs[url].query('Suburb in @sub_list')
    


#df1 = (pd.read_html(urls[0])[5])
#df1.columns = ['Rank', 'Suburb', 'Value']
#df1.drop(df.index[0], inplace=True)

#df2 = (pd.read_html(urls[1])[5])
#df2.columns = ['Rank', 'Suburb', 'Value']
#df2.drop(df.index[0], inplace=True)
  
#check size and 1st few rows
#print(df.shape)
#df.head()    
#comb_df1 =  df1.loc[(df1['Suburb'].isin(inner_sub))|(df1['Suburb'].isin(north_sub))|(df1['Suburb'].isin(south_sub))
 #                 |(df1['Suburb'].isin(east_sub))|(df1['Suburb'].isin(west_sub))]

#comb_df2 =  df2.loc[(df2['Suburb'].isin(inner_sub))|(df2['Suburb'].isin(north_sub))|(df2['Suburb'].isin(south_sub))
#                  |(df2['Suburb'].isin(east_sub))|(df2['Suburb'].isin(west_sub))]

#print(comb_df1.shape, comb_df2.shape)


#comb_df2
#df1 = pd.DataFrame(table_dfs[1])
#df1
#check size and 1st few rows
#print(len(urls))
len(dfs)
#print(df.shape)
#urls[1]
#df2.head()
#type(dfs)


29

In [21]:
#urls = {page_number:'http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=&page={}'.format(page_number) for page_number in list(range(0,2))}

#print("Get urls")

#for url in range(0,len(urls)):
#    print(urls[url], end=', ')

print(dfs.keys())
print(len(dfs.keys()))
df_new = dfs[28]
print(df_new.shape)
#len(urls)
print(list(range(0,len(urls))))

df_new.head()

dict_keys([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28])
29
(8, 3)
[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28]


Rank       Suburb    Value
1  1401    Hughenden  $83,600
2  1402        Bluff  $75,800
3  1403        Kogan  $75,000
4  1404    Greenvale  $74,300
5  1405  Middlemount  $72,300

In [200]:
#filter the rows corresponding to Brisbane suburbs

dfnew = {}
for df in range(0,len(dfs.keys())):
    dfnew[df] = dfs[df].loc[(dfs[df]['Suburb'].isin(sub_list))] 
   

In [6]:
print(dfnew[5].shape)
dfnew[0].head()

(14, 3)


Rank     Suburb       Value
10   10  Willawong  $1,552,100
11   11   New Farm  $1,443,500
15   15  Teneriffe  $1,401,200
16   16    Gumdale  $1,367,000
18   18   Hamilton  $1,357,700

In [9]:
#merge the dataframes vertically 
#brisbane_df = pd.concat([dfnew[0], dfnew[1]], ignore_index=True)
brisbane_df = pd.concat(dfnew.values(), ignore_index=True)
print(brisbane_df.shape)
brisbane_df.head()

(178, 3)


Rank     Suburb       Value
0   10  Willawong  $1,552,100
1   11   New Farm  $1,443,500
2   15  Teneriffe  $1,401,200
3   16    Gumdale  $1,367,000
4   18   Hamilton  $1,357,700

The combined df has 178 rows,  which does not match with the length of sub_list of 192. So we need to findout which suburbs are not in the brisbane_df and mannually add their values 

In [10]:
#check if all inner_sub are in the brisbane_df
#select_values = [31, 22, 30, 25, 64]
#df_inner = brisbane_df.query('Suburb in @inner_sub')
df_north = brisbane_df.query('Suburb in @north_sub')

#print(df_inner.shape, len(inner_sub))
print(df_north.shape, len(north_sub))
#df['one'].tolist()
#df_inner_sub = df_inner['Suburb'].tolist()
df_north_sub = df_north['Suburb'].tolist()

#list(set(inner_sub)-set(df_inner_sub))
list(set(north_sub)-set(df_north_sub))

#all these give common elements in the two lists
#set(df_inner_sub) & set(inner_sub)  
#set(df_inner_sub).intersection(inner_sub)
#bool(set(df_inner_sub).intersection(inner_sub))
#df_inner_sub


(46, 3) 50


['Kalinga', 'Brisbane Airport', 'Myrtletown', 'Eagle Farm']

In [11]:
#read the table in the html webpage: http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=
#df1 = pd.read_html('http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=0')[5]
df = pd.read_html('http://house.speakingsame.com/suburbtop.php?sta=qld&cat=HomePrice&name=&page=0')[5]

#check size and 1st few rows
print(df.shape)
df.head(5)

(51, 3)


0                  1           2
0  Rank             Suburb       Value
1     1     Jollys Lookout  $2,999,900
2     2  Sovereign Islands  $2,619,400
3     3      Mermaid Beach  $2,200,100
4     4     Sunshine Beach  $1,975,100

In [12]:
#rename columns as 'Rank', 'Suburb', 'Value' and drop the first row
df.columns = ['Rank', 'Suburb', 'Value']
df.drop(df.index[0], inplace=True)
  
#check size and 1st few rows
print(df.shape)
df.head()    

(50, 3)


Rank             Suburb       Value
1    1     Jollys Lookout  $2,999,900
2    2  Sovereign Islands  $2,619,400
3    3      Mermaid Beach  $2,200,100
4    4     Sunshine Beach  $1,975,100
5    5    Cooroy Mountain  $1,946,600

18 50 54 28 42


In [13]:
#from df1, extract only the specific suburbs 
inner_df =   df[df['Suburb'].isin(inner_sub)]
north_df =   df[df['Suburb'].isin(north_sub)]
south_df =   df[df['Suburb'].isin(south_sub)]
east_df =   df[df['Suburb'].isin(east_sub)]
west_df =   df[df['Suburb'].isin(west_sub)]

 
print(inner_df.shape,north_df.shape, south_df.shape, east_df.shape, west_df.shape)
#print(north_df.shape)
#print(inner_df)
#print(north_df)
west_df

(5, 3) (4, 3) (3, 3) (5, 3) (5, 3)


Rank           Suburb       Value
19   19  Fig Tree Pocket  $1,337,700
21   21          Chelmer  $1,318,800
24   24         St Lucia  $1,289,300
35   35       Pullenvale  $1,216,500
50   50          Toowong  $1,112,500

In [19]:
#join_df =  df.loc[(df['Suburb'].isin(inner_sub))+(df['Suburb'].isin(north_sub))+(df['Suburb'].isin(south_sub))
#                   +(df['Suburb'].isin(east_sub))+ (df['Suburb'].isin(west_sub))]
#join_df = df[df['Suburb'].str.contains(sub_list, case=False)]
join_df = df.loc[(df['Suburb'].isin(sub_list))]
print(join_df.shape)
join_df


(22, 3)


Rank           Suburb       Value
10   10        Willawong  $1,552,100
11   11         New Farm  $1,443,500
15   15        Teneriffe  $1,401,200
16   16          Gumdale  $1,367,000
18   18         Hamilton  $1,357,700
19   19  Fig Tree Pocket  $1,337,700
20   20          Ransome  $1,329,000
21   21          Chelmer  $1,318,800
22   22         Chandler  $1,317,200
24   24         St Lucia  $1,289,300
26   26         Tennyson  $1,270,600
29   29            Ascot  $1,264,300
31   31          Burbank  $1,261,500
32   32        Hawthorne  $1,244,200
35   35       Pullenvale  $1,216,500
36   36          Bulimba  $1,207,400
38   38       Paddington  $1,202,500
40   40   South Brisbane  $1,175,600
43   43           Hendra  $1,158,300
45   45        Clayfield  $1,139,600
46   46    Highgate Hill  $1,138,800
50   50          Toowong  $1,112,500